### PR TITLE
Alc 97

### PR DESCRIPTION
**Pull request by macroman321**
Please ensure have read our (contribution guidelines](CONTRIBUTING.md) and your
pull request adheres to the following guidelines:

- [ ] Branch name corresponds to Linear ticket number.
- [ ] Added sufficient integration and unit tests.
- [ ] Addressed a single concern in the least number of changed lines as
      possible.
- [ ] Only fixed/added the functionality in question **OR** address wide-spread
      whitespace/style issues, not both.

Thanks!

### CODE REVIEW

File: src/components/TopTokenHolders/index.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+const getRandomColor = () => {
+  var letters = "0123456789ABCDEF";
+  var color = "#";
+  for (var i = 0; i < 6; i++) {
+    color += letters[Math.floor(Math.random() * 16)];
+  }
+  return color;
+};
+
+class TopTokenHolders extends React.Component {
```
Comment (jamespfarrell) [inline src/components/TopTokenHolders/index.tsx]: Have you tried out React hooks, or is there a reason you're going for classes?

**Review by jamespfarrell — COMMENTED** [review]
Looks good besides the use of classes, but I guess Recharts may be forcing this decision?

File: src/components/TopTokenHolders/index.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          4: { offset: 0.2 },
+          12: { offset: 0.3 },
+          14: { offset: 0.4 },
+          15: { offset: 0.5 },
+        },
+      }}
+    />
+  );
+};
+TopTokenHolders.propTypes = {
```
Comment (pygeek) [inline src/components/TopTokenHolders/index.tsx:31]: I believe we don't need prop types, since we already have an interface.

**Review by jamespfarrell — APPROVED** [review]
Looks good

File: src/services/EthplorerService.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        setHasError(true);
+        setLoading(false);
+      });
+  }, [url]);
+  return [response, loading, hasError];
+}
+
+const getHoldersChartData = () => {
+  const [response, loading, hasError] = useFetch(
+    "https://api.ethplorer.io/getTopTokenHolders/0x88acdd2a6425c3faae4bc9650fd7e27e0bebb7ab?apiKey=EK-i88cm-MSc8YqU-bwY7A&limit=100"
```
Comment (pygeek) [inline src/services/EthplorerService.ts:24]: consider placing this in constants and with string interpolation for address

File: src/services/EthplorerService.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  );
+  let googleChartData = [];
+  googleChartData.push(["Top 100 token holders", "Compared to share size"]);
+  if (response) {
+    let sumPercentage = 0.0;
+    // @ts-ignore
```
Comment (pygeek) [inline src/services/EthplorerService.ts:30]: Is there a reason to keep this ignore statement?

File: src/services/EthplorerService.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    response.holders.forEach((holder: { address: any; share: any }) => {
```
Comment (pygeek) [inline src/services/EthplorerService.ts:31]: consider using .map instead

File: src/services/EthplorerService.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      });
+  }, [url]);
+  return [response, loading, hasError];
+}
+
+const getHoldersChartData = () => {
+  const [response, loading, hasError] = useFetch(
+    "https://api.ethplorer.io/getTopTokenHolders/0x88acdd2a6425c3faae4bc9650fd7e27e0bebb7ab?apiKey=EK-i88cm-MSc8YqU-bwY7A&limit=100"
+  );
+  let googleChartData = [];
```
Comment (pygeek) [inline src/services/EthplorerService.ts:26]: Is there a reason why this is called "google"ChartData?

Comment (pygeek) [inline src/services/EthplorerService.ts:26]: Use const

File: src/components/TopTokenHolders/index.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { Chart } from "react-google-charts";
```
Comment (pygeek) [inline src/components/TopTokenHolders/index.tsx:2]: is there a reason why we used this instead of Recharts?

**Review by pygeek — COMMENTED** [review]
Looks good for the most part, just a few feedback items/questions.

**Comment by pygeek** [comment]
I believe you still have a failing test.

File: src/services/EthplorerService.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        setHasError(true);
+        setLoading(false);
+      });
+  }, [url]);
+  return [response, loading, hasError];
+}
+
+const getHoldersChartData = () => {
+  const url =
+    constants.network.ethplorerUrl +
```
Comment (pygeek) [inline src/services/EthplorerService.ts:25]: PS: I meant just creating add constant with entire url using %s string interpolation.

File: src/components/TopTokenHolders/index.tsx (hunk 2 above)
Comment (pygeek) [reply src/components/TopTokenHolders/index.tsx:25]: Think you can remove this and the import of prop-types.

**Review by pygeek — CHANGES_REQUESTED** [review]

**Review by pygeek — APPROVED** [review]

**Review by pygeek — APPROVED** [review]

**Pull request merged**